docs: Updates release notes for app v0.34 (#739)
elastisys/compliantkubernetes: docs/release-notes/ck8s.md

!!!note
    For a more detailed look check out the full [changelog](https://github.com/elastisys/compliantkubernetes-apps/tree/main/changelog).

### v0.34.0

Released 2023-11-21

> [!WARNING]
> **Security Notice(s)**
> - New curl release (CVE-2023-38545 and CVE-2023-38546)
>   New Go release (https://github.com/advisories/GHSA-qppj-fm5r-hxr3 and https://github.com/advisories/GHSA-4374-p667-p6c8)
>   Fix for HTTP/2 Rapid Reset Attack [CVE-2023-44487](https://nvd.nist.gov/vuln/detail/CVE-2023-44487)

> <!-- -->

Reviewer: This does not seem to have been done with the `--output release-notes` option, since this is GitHub flavour.

## Release highlights

> - Increased the default helm timeout to 30 mintues

Reviewer: ```suggestion
```
Why is this a release highlight?

Author: It is from the output of the script.

> - Application developers can now self manage CRDs for SealedSecrets, MongoDB and fluxCD

Reviewer: Yes, we should eventually make the release highlight more like the "this is the core part of this release" type of deal, and I think that would be a good way to communicate bigger changes even to application developers.

I think we need to rework that part in all stages as it's not really clear in the PR template either how that renders down, and currently it only renders into the changelog.

> 

Reviewer 2: You seem to have removed all categorized changes. Was there really no entry worth mentioning in the release notes or did you remove them all for some other reason?

Author: I was under the impression that the reason we have release highlights is so we can just put those on the io page and discard all other categorized changes as they still stay in the apps repository. If that's not the case, I will go through them once again and add some here.

Reviewer: Eventually maybe, but not right now. We need to improve how we write our PRs to make this better first.

!!!note
    For a more detailed look check out the full [changelog](https://github.com/elastisys/compliantkubernetes-apps/tree/main/changelog).

### v0.34.0

Released 2023-11-23

!!! danger "Security Notice(s)"

    - New curl release (CVE-2023-38545 and CVE-2023-38546)
    - New Go release (https://github.com/advisories/GHSA-qppj-fm5r-hxr3 and https://github.com/advisories/GHSA-4374-p667-p6c8)
    - Fix for HTTP/2 Rapid Reset Attack [CVE-2023-44487](https://nvd.nist.gov/vuln/detail/CVE-2023-44487)

### Feature(s)

- Dashboard for visualizing how spread-out pods are across nodes
- Application developers can now self manage CRDs for MongoDB, SealedSecrets and Flux
- Upgrade HNC and expose opt-in propagation

### Improvement(s)

- Update Gatekeeper violation messages
- Add Network Policies for hnc
- Upgrade Ingress-NGINX controller to 1.8.4 and chart to 4.7.3
- Upgrade Falco chart and rework exceptions

### Other(s)

> - Delete namespaces obsolete since v0.20

Reviewer 3: These are just namespaces in sc so to me that does not feel very relevant for the app devs. I suggest that you remove it.

!!!note
    For a more detailed look check out the full [changelog](https://github.com/elastisys/compliantkubernetes-apps/tree/main/changelog).

### v0.34.0

Released 2023-11-23

!!! danger "Security Notice(s)"

>     - New curl release (CVE-2023-38545 and CVE-2023-38546)
>       New Go release (https://github.com/advisories/GHSA-qppj-fm5r-hxr3 and https://github.com/advisories/GHSA-4374-p667-p6c8)
>       Fix for HTTP/2 Rapid Reset Attack [CVE-2023-44487](https://nvd.nist.gov/vuln/detail/CVE-2023-44487)

Reviewer 2: Unfortunately currently multiline notices aren't preserved in these blockquotes. In the [original pull request](https://github.com/elastisys/compliantkubernetes-apps/pull/1829) it looks fine which is problematic since it doesn't when it's rendered as a changelog.

I created an issue for it https://github.com/elastisys/releaser/issues/14. One way we could solve it would be to translate newlines into separate notices.

```suggestion
    - New curl release (CVE-2023-38545 and CVE-2023-38546)
    - New Go release (https://github.com/advisories/GHSA-qppj-fm5r-hxr3 and https://github.com/advisories/GHSA-4374-p667-p6c8)
    - Fix for HTTP/2 Rapid Reset Attack [CVE-2023-44487](https://nvd.nist.gov/vuln/detail/CVE-2023-44487)
```

Author: Ah.. thanks for the catch. Also thanks for opening the issue.

### Feature(s)

- Dashboard for visualizing how spread-out pods are across nodes
- Application developers can now self manage CRDs for MongoDB and SealedSecrets
- Upgrade HNC and expose opt-in propagation

### Improvement(s)

- Update Gatekeeper violation messages
- Add Network Policies for hnc

> - Add support for Flux with self-managed CRDs

Reviewer: Maybe merge the Flux entry with the other user dev CRDs.

> ### Feature(s)
> 
> - sc apps: Dashboard for visualizing how spread-out pods are across nodes
> - Application developer self managed CRDs
> - upgrade hnc and expose opt-in propagation
> 
> ### Improvement(s)
> 
> - apps: update gatekeeper violation messages
> - apps: add networkpolicies for hnc
> - Added support for Flux with self-managed CRDs
> - apps: upgrade ingress-nginx controller to 1.8.4 and chart to 4.7.3
> - apps: upgrade falco chart and rework exceptions
> 
> ### Other(s)
> 
> - clean-up - migration: Delete namespaces obsolete since v0.20

Reviewer 2: Just some cleanup of the messages. We probably don't need to mention `apps:` prefix looks redundant for an compliantkubernetes-apps release. :smile: 

```suggestion
### Feature(s)

- Dashboard for visualizing how spread-out pods are across nodes
- Application developer self managed CRDs
- Upgrade HNC and expose opt-in propagation

### Improvement(s)

- Update Gatekeeper violation messages
- Add Network Policies for hnc
- Add support for Flux with self-managed CRDs
- Upgrade Ingress-NGINX controller to 1.8.4 and chart to 4.7.3
- Upgrade Falco chart and rework exceptions

### Other(s)

- Delete namespaces obsolete since v0.20
```

Author: Sure !!

!!!note
    For a more detailed look check out the full [changelog](https://github.com/elastisys/compliantkubernetes-apps/tree/main/changelog).

### v0.34.0

Released 2023-11-23

!!! danger "Security Notice(s)"

    - New curl release (CVE-2023-38545 and CVE-2023-38546)
      New Go release (https://github.com/advisories/GHSA-qppj-fm5r-hxr3 and https://github.com/advisories/GHSA-4374-p667-p6c8)
      Fix for HTTP/2 Rapid Reset Attack [CVE-2023-44487](https://nvd.nist.gov/vuln/detail/CVE-2023-44487)

### Feature(s)

- sc apps: Dashboard for visualizing how spread-out pods are across nodes
- Application developer self managed CRDs

> - Disk performance dashboard

Reviewer: ```suggestion
```
Not sure this is dev facing enough.

- upgrade hnc and expose opt-in propagation

### Improvement(s)

> - apps: Upgrade kube-prometheus-stack to v49.2.0
> - apps: update gatekeeper violation messages
> - apps: add networkpolicies for hnc
> - Added support for Flux with self-managed CRDs
> - apps: upgrade ingress-nginx controller to 1.8.4 and chart to 4.7.3
> - apps: upgrade falco chart and rework exceptions
> - increase helm timeout
>   - Increased the default helm timeout to 30 mintues

Reviewer: ```suggestion
- apps: update gatekeeper violation messages
- apps: add networkpolicies for hnc
- Added support for Flux with self-managed CRDs
- apps: upgrade ingress-nginx controller to 1.8.4 and chart to 4.7.3
- apps: upgrade falco chart and rework exceptions
```
